formfiller/src/main/java/com/diyasylum/formfiller/mappings/i589/I589ApplicationMapper.java

```diff
@@ -0,0 +1,27 @@
+package com.diyasylum.formfiller.mappings.i589;
+
+import com.diyasylum.formfiller.application.models.*;
+import com.diyasylum.formfiller.mappings.AbstractMapper;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+public class I589ApplicationMapper extends AbstractMapper<I589Application> {
+  public static final String FORM_URL =
+      "https://www.uscis.gov/system/files_force/files/form/i-589.pdf?download=1";
+  public static final String SUPPORTED_FORM_REVISION = "Form I-589 (Rev. 05/16/17) N";
+
+  private static Map<String, Function<I589Application, String>> fieldMapping;
+
+  @Override
+  public Map<String, Function<I589Application, String>> getFormMapper() {
+    if (fieldMapping == null) {
+      fieldMapping = new HashMap<>();
+      fieldMapping.putAll(
+          new UsTravelHistoryMapper().composeMapping(I589Application::getUsTravelHistory));
+      fieldMapping.putAll(
```

Comment:
When we have multiple supplements, the PDF path string will no longer be in 1-1 correspondence with text, since the same field path will appear on all the supplements. How will we eventually handle this?

Reply:
I don't think 100% understand. .

My thinking is each supplement would get it's own mapping package. So both can access the data however they want. 

If both the form and the supplement need the same data they probably have different conditions for needing to actually use that data.

So for name. Main form would use it if the name was fewer than x characters and supplement would use it if it was greater than or equal to x characters.